lib/maverick/api/initializer.ex

```diff
@@ -1,168 +1,145 @@
+                var!(conn),
+                response,
+                unquote(success),
+                unquote(error)
+              )
+            rescue
+              exception ->
+                %{tag: tag, handler: {mod, func, args}} = Maverick.Exception.fallback(exception)
+
+                Logger.info(fn ->
+                  "#{inspect(exception)} encountered processing request #{inspect(var!(conn))}; falling back to #{
+                    tag
+                  }"
+                end)
+
+                response = apply(mod, func, args)
+
+                var!(conn)
+                |> Plug.Conn.put_resp_content_type("application/json", nil)
+                |> Plug.Conn.send_resp(Maverick.Exception.error_code(exception), response)
+            end
           end
         end
-      end
 
-    quote location: :keep, bind_quoted: [contents: contents], do: contents
+      result
+    end
   end
 
-  defp variablize_path(path) do
-    Enum.map(path, fn element ->
-      case element do
-        {:variable, variable} ->
-          variable
-          |> String.to_atom()
-          |> Macro.var(__MODULE__)
+  def to_request(conn) do
+    %Maverick.Request{
+      body: nil,
+      body_params: conn.params,
+      headers: conn.req_headers |> Enum.into(%{}),
+      host: conn.host,
+      method: conn.method |> to_string(),
+      params: conn.params,
+      path: conn.request_path,
+      path_params: conn.path_params,
+      port: conn.port,
+      query_params: conn.query_params,
+      raw_path: conn.request_path,
+      remote_ip: conn.remote_ip,
+      scheme: conn.scheme,
+      socket: nil,
+      version: nil
+    }
+  end
 
-        _ ->
-          element
-      end
-    end)
+  def wrap_response(conn, {:ok, headers, response}, success, _error) do
+    conn
+    |> Plug.Conn.put_resp_content_type("application/json", nil)
+    |> add_headers(headers)
+    |> Plug.Conn.send_resp(success, Jason.encode!(response))
   end
 
-  defp path_var_map(path) do
-    entries =
-      Enum.reduce(path, [], fn element, acc ->
-        case element do
-          {:variable, variable} ->
-            value = variable |> String.to_atom() |> Macro.var(__MODULE__)
-            [{variable, value} | acc]
+  def wrap_response(conn, {:error, error_message}, _success, error) do
+    response =
+      %{error_code: error, error_message: error_message}
+      |> Jason.encode!()
 
-          _ ->
-            acc
-        end
-      end)
+    conn
+    |> Plug.Conn.put_resp_content_type("application/json", nil)
+    |> Plug.Conn.send_resp(error, response)
+  end
 
-    {:%{}, [], entries}
+  def wrap_response(conn, response, success, _error) do
+    conn
+    |> Plug.Conn.put_resp_content_type("application/json", nil)
+    |> Plug.Conn.send_resp(success, Jason.encode!(response))
+  end
+
+  def decode_arg_type(conn, :request) do
+    Maverick.Api.Initializer.to_request(conn)
+  end
+
+  def decode_arg_type(conn, _) do
+    Map.get(conn, :params)
+  end
```

Comment:
If this function is never imported via `_using_` itself and is only really referenced by a fully qualified module name from within an imported function (line 57) why not just reference the internal function here (assuming we don't move that function to the request module or just get rid of the `Request` altogether)?
```suggestion
  def decode_arg_type(conn, :request) do
    to_request(conn)
  end

  def decode_arg_type(conn, _) do
    Map.get(conn, :params)
  end
```

Reply:
planning on moving these functions to another location, just not sure where yet.